### PR TITLE
Add an item's basic model

### PR DESCRIPTION
**Pull request by melissaharijanto**
An item can take in 5 things: 
- A name (String)
- A description (String)
- A quantity (int)
- A set of tags (which will probably be implemented much much later, so if it hinders you in any of the implementations, please feel free to remove!)

What I implemented for these are mostly just the `equals` methods as well as the getter methods.

### CODE REVIEW

File: src/main/java/tracko/model/items/Quantity.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Represents the quantity of an item.
+ */
+public class Quantity {
+    public final int quantity;
+
+    /**
+     * Constructs a {@code Quantity}.
+     * @param quantity The quantity of an item.
+     */
+    public Quantity(int quantity) {
```
Comment (donljh) [inline src/main/java/tracko/model/items/Quantity.java:19]: I think Quantity can be validated as not being a negative integer here

**Review by donljh — CHANGES_REQUESTED** [review]

File: src/main/java/tracko/model/items/Quantity.java (hunk 1 above)
Comment (melissaharijanto) [reply src/main/java/tracko/model/items/Quantity.java:19]: That's a good catch! Thank you, I fixed it

File: src/main/java/tracko/model/items/Quantity.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Quantity)) {
+            return false;
+        }
+
+        Quantity otherPair = (Quantity) other;
```
Comment (donljh) [inline src/main/java/tracko/model/items/Quantity.java:48]: I think to make the naming less confusing, this should be otherQuantity?

**Review by donljh — APPROVED** [review]
Overall, looks good! Can maybe change the naming in the equals method, but otherwise everything looks ok

**Review by Jiacheng-y — APPROVED** [review]
Generally looks good to merge! 

I just have some points that I wish to point out for the current design: 

- we will need to change our UG and DG to include quantity and description for items (I'll open up issues for this. I think atm only the parts for addi is affected)
- are we mandating a description for each item? 

Apart from that, I'm good with this! Thank you Melissa!

**Pull request merged**